cmd/zed/lake/index/create.go

```diff
@@ -37,136 +43,96 @@ requires specifying the key and output file name. For example:
-	c := &CreateCommand{Command: parent.(*Command).Command}
-	f.StringVar(&c.root, "R", os.Getenv("ZED_LAKE_ROOT"), "root location of zar archive to walk")
-	f.BoolVar(&c.ensure, "ensure", false, "ensures that index rules are applied to all chunks")
-	f.StringVar(&c.keys, "k", "key", "one or more comma-separated key fields")
-	f.IntVar(&c.framesize, "f", 32*1024, "minimum frame size used in microindex file")
-	f.StringVar(&c.inputFile, "i", "_", "input file relative to each zar directory ('_' means archive log file in the parent of the zar directory)")
-	f.BoolVar(&c.noapply, "noapply", false, "create index rules but do not apply them to the archive")
-	f.StringVar(&c.outputFile, "o", "index.zng", "name of microindex output file (for custom indexes)")
-	//f.BoolVar(&c.displayer.quiet, "q", false, "don't print progress on stdout")
-	f.StringVar(&c.zql, "z", "", "zql for custom indexes")
+	c := &CreateCommand{lake: parent.(*Command).Command}
+	f.BoolVar(&c.commit, "commit", false, "commit added index rule if successfully written")
+	f.IntVar(&c.framesize, "framesize", 32*1024, "minimum frame size used in microindex file")
+	f.StringVar(&c.keys, "k", "key", "one or more comma-separated key fields (for zed script index rules only)")
+	f.StringVar(&c.name, "n", "", "name of index rule (for zed script index rules only)")
+	f.StringVar(&c.zed, "zed", "", "zed script for rule")
+	c.outputFlags.DefaultFormat = "lake"
+	c.outputFlags.SetFlags(f)
 	c.procFlags.SetFlags(f)
+	c.CommitFlags.SetFlags(f)
 	return c, nil
 }
 
 func (c *CreateCommand) Run(args []string) error {
-	return errors.New("issue #2532")
-	/*
-		defer c.Cleanup()
-		if err := c.Init(&c.procFlags); err != nil {
-			return err
-		}
-		if len(args) == 0 && c.zql == "" && !c.ensure {
-			return errors.New("unless -ensure is specified, one or more indexing patterns must be specified")
-		}
-		if c.root == "" {
-			return errors.New("a directory must be specified with -R or ZED_LAKE_ROOT")
-		}
-		if _, err := rlimit.RaiseOpenFilesLimit(); err != nil {
-			return err
-		}
+	ctx, cleanup, err := c.lake.Init(&c.procFlags, &c.outputFlags)
+	if err != nil {
+		return err
+	}
+	defer cleanup()
 
-		lk, err := lake.OpenLake(c.root, nil)
-		if err != nil {
-			return err
-		}
+	if len(args) == 0 && c.zed == "" {
+		return errors.New("one or more index rule patterns must be specified")
+	}
+
+	if _, err := rlimit.RaiseOpenFilesLimit(); err != nil {
+		return err
+	}
 
-		ctx, cancel := signalctx.New(os.Interrupt)
-		defer cancel()
+	root, err := c.lake.Open(ctx)
+	if err != nil {
+		return err
+	}
+
+	rules, err := c.createXRules(ctx, root, args)
+	if err != nil {
+		return err
+	}
 
-		defs, err := c.addRules(ctx, lk, args)
+	if !c.lake.Quiet {
+		w, err := c.outputFlags.Open(ctx)
 		if err != nil {
 			return err
 		}
 
-		if c.noapply {
-			return nil
-		}
-
-		if c.ensure {
-			defs, err = lk.ReadDefinitions(ctx)
+		m := zson.NewZNGMarshaler()
```

Comment:
This should use root.ScanXRules.  There shouldn't be a need to have a marshaling loop like this at the command level.